client/src/connection/ssh/index.ts

```diff
@@ -15,28 +14,32 @@ export interface Config {
   saspath: string;
   sasOptions: string[];
   port: number;
-  privateKeyPath: string;
 }
 
 export function getSession(c: Config): Session {
+  if (!process.env.SSH_AUTH_SOCK) {
+    throw new Error("SSH_AUTH_SOCK not set. Check Environment Variables.");
+  }
+
   if (!sessionInstance) {
-    sessionInstance = new SSHSession(c);
+    sessionInstance = new SSHSession();
   }
+  sessionInstance.config = c;
   return sessionInstance;
 }
 
 export class SSHSession implements Session {
   private conn: Client;
   private stream: ClientChannel | undefined;
-  private config: Config;
+  public config: Config;
```

Comment:
I feel like it would be better to leave this private and create a Getter. This also makes more sense now that the config is an optional parameter in the constructore.

Reply:
@JosephDHenry  fixed, see updated.